compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java

```diff
@@ -183,9 +183,8 @@ void instrumentPackage(BIRPackage pkg) {
                 }
                 if (serviceName == null) {
                     String basePath = this.svcAttachPoints.get(typeDef.name);
-                    serviceName = Objects.requireNonNullElseGet(basePath, () ->
-                            pkg.packageID.orgName.value + "_" + pkg.packageID.name.value + "_svc_" +
-                            defaultServiceIndex++);
+                    serviceName = basePath == null ? pkg.packageID.orgName.value + "_" + pkg.packageID.name.value +
+                            "_svc_" + defaultServiceIndex++ : Utils.unescapeBallerina(basePath);
```

Comment:
```suggestion
                    serviceName = basePath == null ? pkg.packageID.orgName.getValue() + "_" + pkg.packageID.name.getValue() +
                            "_svc_" + defaultServiceIndex++ : Utils.unescapeBallerina(basePath);
```
Shall we use the getter method to get the name strings? The field `value` in the class `Name.java` should be private (at least in future). A [TODO](https://github.com/ballerina-platform/ballerina-lang/blob/c6f9327976c7eb64dd2d93fa75a5b4d053c7cdfa/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/Name.java#L25) has already been added for that.